server/svix-server/tests/utils/mod.rs

```diff
@@ -242,36 +242,45 @@ pub struct TestReceiver {
 
 impl TestReceiver {
     pub fn start(resp_with: axum::http::StatusCode) -> Self {
-        let listener = TcpListener::bind("127.0.0.1:0").unwrap();
-        let endpoint = format!("http://{}/", listener.local_addr().unwrap());
-
-        let (tx, data_recv) = mpsc::channel(32);
-        let (header_tx, header_recv) = mpsc::channel(32);
-
-        let jh = tokio::spawn(async move {
-            axum::Server::from_tcp(listener)
-                .unwrap()
-                .serve(
-                    axum::Router::new()
-                        .route(
-                            "/",
-                            axum::routing::post(test_receiver_route).get(test_receiver_route),
-                        )
-                        .layer(axum::extract::Extension(tx))
-                        .layer(axum::extract::Extension(header_tx))
-                        .layer(axum::extract::Extension(resp_with))
-                        .into_make_service(),
-                )
-                .await
-                .unwrap();
-        });
-
-        TestReceiver {
-            endpoint,
-            jh,
-            data_recv,
-            header_recv,
-        }
+        generate_axum_server(resp_with, None)
+    }
+
+    pub fn start_on_port(resp_with: axum::http::StatusCode, port: i32) -> Self {
+        generate_axum_server(resp_with, Some(port))
+    }
+}
+
+fn generate_axum_server(resp_with: axum::http::StatusCode, port: Option<i32>) -> TestReceiver {
```

Comment:
This is still an open topic (from the discussion with James too), just make the behavior controllable (heck, we can even just support passing a custom `Fn`) so that you can make it succeed/fail whenever you want. It can be as simple as calling a function that changes the "return status code" from 200 to 403.

Reply:
The hiccup here is, it does not seem that both the struct TestReceiver and an `axum.layer` consumption can share some structure indicating status code.

Reply:
Mutex, RwLock, Arc, etc. There are many tools for sharing structures.